src/resolvers/api/getNode.js

```diff
@@ -0,0 +1,147 @@
+const _get = require('lodash.get')
+
+const { query } = require('../utils')
+
+const getNode = async (_, args) => {
+  const { urbitId } = _get(args, 'input.urbitId') || null
+
+  let nodeType = null
+
+  if (urbitId.length === 4) {
+    nodeType = 'GALAXY'
+  } else if (urbitId.length === 7) {
+    nodeType = 'STAR'
+  } else if (urbitId.length === 14) {
+    nodeType = 'PLANET'
+  }
+
+  const getNumOwnersQuery = `select count(*) from pki_events where node_id = '${urbitId}' and event_type_id = '1';`
+  const getNumOwnersResponse = await query(getNumOwnersQuery)
+
+  const numOwners = parseInt(_get(getNumOwnersResponse, 'rows[0].count')) || 1
+
+  const sponsors = []
+
+  // sponsor
+  const getSponsorQuery = `select sponsor_id from pki_events where node_id = '${urbitId}' order by time desc limit 1;`
+  const getSponsorResponse = await query(getSponsorQuery)
+  const getSponsorResponseRow = _get(getSponsorResponse, 'rows[0]') || []
+
+  const sponsor = getSponsorResponseRow ? _get(getSponsorResponseRow, 'sponsor_id') || null : null
+
+  sponsors.push(sponsor)
+
+  // sponsor's sponsor
+  const getSponsorsSponsorQuery = `select sponsor_id from pki_events where node_id = '${sponsor}' order by time desc limit 1;`
+  const getSponsorsSponsorResponse = await query(getSponsorsSponsorQuery)
+  const getSponsorsSponsorResponseRow = _get(getSponsorsSponsorResponse, 'rows[0]') || []
+  const sponsorsSponsor = getSponsorsSponsorResponseRow ? _get(getSponsorsSponsorResponseRow, 'sponsor_id') || null : null
+
+  if (sponsorsSponsor) {
+    sponsors.push(sponsorsSponsor)
+  }
+
+  // status
+  let statusId
+  let getStatusIdQuery = `select count(result) from radar where ship_name = '${urbitId}';`
+  let getStatusIdResponse
+  try {
+    getStatusIdResponse = await query(getStatusIdQuery)
+
+    if (_get(getStatusIdResponse, 'rows[0].count') === '0') {
+      getStatusIdQuery = `select count(*) from pki_events where node_id = '${urbitId}' and event_type_id = '6';`
+      getStatusIdResponse = await query(getStatusIdQuery)
+
+      if (_get(getStatusIdResponse, 'rows[0].count') === '0') {
+        getStatusIdQuery = `select count(*) from pki_events where node_id = '${urbitId}' and event_type_id = '7';`
+        getStatusIdResponse = await query(getStatusIdQuery)
+
+        if (_get(getStatusIdResponse, 'rows[0].count') === '0') {
+          getStatusIdQuery = `select count(*) from pki_events where node_id = '${urbitId}' and address = '0x86cd9cd0992f04231751e3761de45cecea5d1801' or address = '0x8c241098c3d3498fe1261421633fd57986d74aea';`
+          getStatusIdResponse = await query(getStatusIdQuery)
+
+          if (_get(getStatusIdResponse, 'rows[0].count') === '0') {
+            statusId = 1
+          } else {
+            statusId = 2
+          }
+        } else {
+          statusId = 3
+        }
+      } else {
+        statusId = 4
+      }
+    } else {
+      statusId = 5
```

Comment:
These are [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) and should be replaced with constants that refer to them by name.

Reply:
Good catch! I had these in here as a placeholder at one point and forgot to swap them out for constants. Fixed.